### PR TITLE
Day3

### PR DESCRIPTION
**Pull request by LauraMihalca**

### CODE REVIEW

File: project/.gitignore [hunk 1]
```diff
@@ -0,0 +1,8 @@
+login.pdf
+evaluations.pdf
+New evaluation.pdf
+node_modules
+package.json
+.idea
+images/*png
```
Comment (softvision-alexcarcea) [inline project/.gitignore:7]: Time for a little `git` talk: today's episode - `.gitignore`
Normally, you'll want your `.gitignore` file to include all files that are not relevant to the actual source code of your project. These typically include IDE settings files, user preferences files, external dependencies that your project is configured to bring in during build time, files output by the build process etc.

For your typical web project, I'd normally expect assets such as images to be part of the repository, such that any developer working on the project has the exact same images available when setting up their environment. Now, you could also store images or other similar resources in a separate repository and use that to serve files into your project, but that complicates things too much for the purpose of this comment.

So, bottom line, including these last two lines in your `.gitignore` file is _probably_ fine for the purposes of this training project (and it saves us, reviewers, from the trouble of having to scroll through stuff we don't really care about all that much), but I do want you to be aware of what's happening under the hood when you do something like this.

File: project/assets/common.less [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  font-weight: bold;
+  line-height: 105px;
+}
+
+button {
+  border: none;
+  background-color: rgb(51, 153, 255);
+  color: rgb(255, 255, 255);
+  font-family: roboto;
+  box-shadow: 0px 10px 35px -9px rgba(204,194,204,1);
```
Comment (softvision-alexcarcea) [inline project/assets/common.less:16]: Any particular reason why you chose `rgba` format here? I see that you prefer `rgb` otherwise.

File: project/assets/common.less [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  line-height: 30px;
+  width: 25%;
+  font-size: 20px;
+  margin-top: 42%;
```
Comment (softvision-alexcarcea) [inline project/assets/common.less:20]: Hmm... so every button in your app will _always_ have a top margin of `42%` and a left margin of `33%`? For all devices and screen sizes? How certain are you? :stuck_out_tongue:

File: project/assets/login.less [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+  }
+}
+
+.login-section-1{
+  background-color: rgb(255, 255, 255);
+  padding: 70px;
+}
+
+.login-section-text{
+  padding: 115px;
```
Comment (softvision-alexcarcea) [inline project/assets/login.less:23]: Quite a bit of padding here. Why do we need so much?

File: project/assets/login.less [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+}
+
+.login-section-2{
+  padding: 30px 30px;
```
Comment (softvision-alexcarcea) [inline project/assets/login.less:27]: Can we simplify this? If so, how?

File: project/assets/new-evaluation.less [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+  margin-top: 4%;
+}
+
+
+option, select{
+  font-size: x-large;
+  color: rgb(77, 77, 77);
+  width: 7%;
+  text-align: center;
+  margin-left: 4%;
```
Comment (softvision-alexcarcea) [inline project/assets/new-evaluation.less:64]: Could we group these into a single `margin` property?

File: project/login.html [hunk 7]
```diff
@@ -1,59 +1,85 @@
...
+
+
 <body>
 
   <header>
+      <div class="grid">
+        <div class="login-form">
+              <form>
+                  <div class="login-form-header">
+                    <h3>Interview Feedback</h3>
```
Comment (softvision-alexcarcea) [inline project/login.html:23]: Normally, the recommendation is to always include all preceding heading levels, so as to confer structure and semantic meaning to your document. Not saying you should never, ever, absolutely never have something like what you have here, but just curious whether you chose the `<h3>` for specific reasons or simply because of default styling reasons.

File: project/login.html [hunk 8]
```diff
@@ -1,59 +1,85 @@
...
+                  </div>
+                  <div class="login-form-items">
+                      <input type="text" placeholder="Username">
```
Comment (softvision-alexcarcea) [inline project/login.html:26]: I wonder if you guys went into `form`s and `input`s in more depth. If you were to submit this form, what information would you be passing to the receiving endpoint and how?

File: project/login.html [hunk 9]
```diff
@@ -1,59 +1,85 @@
...
+                      <input type="password" placeholder="Password">
+                      <button class="button">Login</button>
```
Comment (softvision-alexcarcea) [inline project/login.html:28]: By default, if a `form` doesn't contain other, more specific controls for submitting the data, the first `button` inside the form will trigger data submission. However, since there are other controls that are better suited for this purpose, I'd recommend using one of those, for clarity.

File: project/login.html [hunk 10]
```diff
@@ -1,59 +1,85 @@
...
-      <p>Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.</p>
-    </div>
-</section>
+</body>
 
-<footer style="margin-top:10%; float:right;">
-  <p>
-    Copyright@Softvision 2017
-  </p>
+<footer>
```
Comment (softvision-alexcarcea) [inline project/login.html:79]: Does this element actually show up on page? In all browsers? What does the [W3C HTML Validator](https://validator.w3.org/) have to say about this markup?

File: project/newEvaluation.html [hunk 11]
```diff
@@ -1,472 +1,489 @@
...
 
 
-      <section>
-        <h2>Should the candidate be hired?</h2>
-        <textarea rows="6" placeholder="The type of project that is suitable for the candidate
-Is guidance required for the candidate" reqired></textarea>
-      </section>
+          <section>
+                <h2>Should the candidate be hired?</h2>
+                <textarea rows="6" placeholder="The type of project that is suitable for the candidate
```
Comment (softvision-alexcarcea) [inline project/newEvaluation.html:77]: Just curious whether this newline gets properly rendered in the resulting control.

File: project/newEvaluation.html [hunk 12]
```diff
@@ -1,472 +1,489 @@
...
-                <option value="3">3</option>
-              </select>
-          </li>
-      </fieldset>
+          <fieldset>
+            <legend>OOP, Design Patterns</legend>
+            <ul>
+              <li>
+                <label for="classes">Classes</label>
+                <select name="classes" id="classes">
```
Comment (softvision-alexcarcea) [inline project/newEvaluation.html:97]: I assume that you're specifying an ID for these so that you can later hook up the corresponding `<label>` to this control? If that's the case, can you tell me another option of doing the same that's less verbose?

File: project/newEvaluation.html [hunk 13]
```diff
@@ -1,472 +1,489 @@
...
+                  <label for="eventh">Event Handling</label>
+                  <select name="eventh" id="eventh">
+                    <option value="0">0</option>
+                    <option value="1">1</option>
+                    <option value="2">2</option>
+                    <option value="3">3</option>
+                    </select>
+                </li>
+                <li>
+                  <label for="prot">Prototype & OOP</label>
```
Comment (softvision-alexcarcea) [inline project/newEvaluation.html:359]: Guess what I'm gonna comment about this line of code. If you've guessed it, see other occurrences throughout this file, as well. :wink:

**Review by softvision-alexcarcea — APPROVED** [review]

File: project/.gitignore (hunk 1 above)
Comment (LauraMihalca) [reply project/.gitignore:7]: Yes, I was aware of what I've done and thank you for the explanation.

File: project/assets/common.less (hunk 2 above)
Comment (LauraMihalca) [reply project/assets/common.less:16]: Hmm... I used a web tool for composing this shadow...but I'll change it for rgb

File: project/login.html (hunk 8 above)
Comment (LauraMihalca) [reply project/login.html:26]: Well, I should give each input a name=something in order to be able to submit the information. I omitted that because I was only concerned about the style last week. I know it's a bad practice because I could have forgotten about them, but I realize it just now, so thank you :D

File: project/login.html (hunk 9 above)
Comment (LauraMihalca) [reply project/login.html:28]: Do you say to use --- input type="submit" value="SUBMIT" ---?

File: project/newEvaluation.html (hunk 11 above)
Comment (LauraMihalca) [reply project/newEvaluation.html:77]: Do you mean the two lines from placeholder ?

File: project/assets/common.less (hunk 2 above)
Comment (softvision-alexcarcea) [reply project/assets/common.less:16]: Alright, if you really, really want to! I was just asking out of curiosity, really. :)

File: project/assets/login.less (hunk 4 above)
Comment (LauraMihalca) [reply project/assets/login.less:23]: Is not ok to set paddings very large ? I set it this big because it seemed to look good..

File: project/assets/login.less (hunk 5 above)
Comment (LauraMihalca) [reply project/assets/login.less:27]: padding: 30px; :D

File: project/assets/new-evaluation.less (hunk 6 above)
Comment (LauraMihalca) [reply project/assets/new-evaluation.less:64]: margin: 1% 5% 0 4%;

File: project/login.html (hunk 7 above)
Comment (LauraMihalca) [reply project/login.html:23]: Just for the default styling

File: project/newEvaluation.html (hunk 13 above)
Comment (LauraMihalca) [reply project/newEvaluation.html:359]: Is it because of the & ?

File: project/login.html (hunk 9 above)
Comment (softvision-alexcarcea) [reply project/login.html:28]: `<button type="submit">` or `<input type="submit">` should work just as well. This is more for clarity of the code than anything else, although I've seen cross-browser issues around handling of buttons and inputs (especially in older versions of IE), hence my initial comment.

File: project/newEvaluation.html (hunk 11 above)
Comment (softvision-alexcarcea) [reply project/newEvaluation.html:77]: Exactly. :)

File: project/assets/login.less (hunk 4 above)
Comment (softvision-alexcarcea) [reply project/assets/login.less:23]: It isn't wrong to do so, by any means. I just found it strange that it's such a large value, but if you feel it looks good, I won't be one to argue. :)

File: project/login.html (hunk 7 above)
Comment (softvision-alexcarcea) [reply project/login.html:23]: Then, there might be better options, should you choose to look for them. :wink:

File: project/newEvaluation.html (hunk 13 above)
Comment (softvision-alexcarcea) [reply project/newEvaluation.html:359]: It is because of the `&`. And what do we need to do about the `&`? :stuck_out_tongue:

Comment (LauraMihalca) [reply project/newEvaluation.html:359]: I have to write its entity name which I cannot write here because it will transform it into simple & :))

File: project/newEvaluation.html (hunk 11 above)
Comment (LauraMihalca) [reply project/newEvaluation.html:77]: Well..they look good...

File: project/newEvaluation.html (hunk 13 above)
Comment (softvision-alexcarcea) [reply project/newEvaluation.html:359]: That's because GitHub uses Markdown syntax by default, which is, in a sense, a superset of HTML, and thus automatically encodes your entities to their corresponding character value. What you _can_ do, though, is use backtick (*\`*) characters around the string to escape the actual content, so that it's displayed like so: `&amp;`. Fun fact: you can use the `Preview` tab above to see what the end result will look like. :wink:

Comment (LauraMihalca) [reply project/newEvaluation.html:359]: Thaaaank you!!!
